apps/web/src/app/(default)/for-studenter/arrangementer/page.tsx

```diff
@@ -1,11 +1,144 @@
+  };
+
+  if (params.open === "true" && params.past === "true") {
+    validParams.open = undefined;
+    validParams.past = undefined;
+  } else {
+    validParams.open = params.open === "true" ? "true" : undefined;
+    validParams.past = params.past === "true" ? "true" : undefined;
+  }
+
+  return validParams;
+}
+
+function getDateInterval(params: SearchParams) {
+  const currentDate = new Date();
+
+  const past = params.past === "true" ? true : false;
+  const hideThisWeek = params.thisWeek === "false" ? true : false;
+  const hideNextWeek = params.nextWeek === "false" ? true : false;
+  const hideLater = params.later === "false" ? true : false;
+
+  if (past) {
+    return [{ end: currentDate }];
+  }
+  if (!hideThisWeek && !hideNextWeek && !hideLater) return [{ start: subMinutes(currentDate, 30) }];
+  if (hideThisWeek && !hideNextWeek && !hideLater) return [{ start: nextMonday(currentDate) }];
+  if (!hideThisWeek && hideNextWeek && !hideLater)
+    return [
+      { start: subMinutes(currentDate, 30), end: nextMonday(currentDate) },
+      { start: nextMonday(nextMonday(currentDate)) },
+    ];
+  if (!hideThisWeek && !hideNextWeek && hideLater)
+    return [{ start: subMinutes(currentDate, 30), end: nextMonday(nextMonday(currentDate)) }];
+  if (!hideThisWeek && hideNextWeek && hideLater)
+    return [{ start: subMinutes(currentDate, 30), end: nextMonday(currentDate) }];
+  if (hideThisWeek && !hideNextWeek && hideLater)
+    return [{ start: nextMonday(currentDate), end: nextMonday(nextMonday(currentDate)) }];
+  if (hideThisWeek && hideNextWeek && !hideLater)
+    return [{ start: nextMonday(nextMonday(currentDate)) }];
+
+  return undefined;
+}
+
+export default async function Page({ searchParams }: { searchParams?: SearchParams }) {
+  const validParams = validateParams(searchParams);
+
+  const query = generateQuery(validParams);
+  const { numThisWeek, numNextWeek, numLater, happenings } = await fetchFilteredHappening(query);
+
+  if (!happenings)
+    return (
+      <Callout type="danger" className="mx-auto my-8 w-full max-w-lg">
+        <p className="font-bold">En feil har oppstått...</p>
+        <p>Prøv igjen senere</p>
+      </Callout>
+    );
+
+  if (validParams.order === "ASC") happenings.reverse();
 
-export default function HomePage() {
   return (
     <Container>
-      <section className="flex flex-col gap-5 rounded-md border p-5">
-        <Events />
+      <section>
+        <div>
+          <EventFilterBar params={validParams} />
+        </div>
+        <div className="p-3">
+          <EventSearchAndOrderBar params={validParams} />
+        </div>
+        <div className="flex">
+          <div className="w-1/4">
+            <EventDateFilterSidebar
+              params={validParams}
+              numOfEvents={{ numThisWeek, numNextWeek, numLater }}
+            />
+          </div>
+          <div className="w-3/4">
+            <EventsView happenings={happenings} />
+          </div>
```

Comment:
Jeg hadde gjort det sånn

```suggestion
        <div className="flex flex-col md:flex-row">
          <div className="w-full md:max-w-[200px]">
            <EventDateFilterSidebar
              params={validParams}
              numOfEvents={{ numThisWeek, numNextWeek, numLater }}
            />
          </div>
          <div>
            <EventsView happenings={happenings} />
          </div>
```

Og i `<EventsView />`:
```
<div className="w-full"> // La til w-full
  {happenings.map((event) => (
    <ul key={event._id} className="py-1">
      <CombinedHappeningPreview happening={event} />
    </ul>
  ))}
</div>
```